### PR TITLE
Fix failure to run eslint on win platform

### PR DESCRIPTION
**Pull request by justinbhopper**
Fixes errors when running `lerna run lint` on a win-platform.

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+registry=https://registry.npmjs.org
```
Comment (adidahiya) [inline .npmrc:1]: we use yarn, so I don't think this is necessary

File: .eslintignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 node_modules
 dist
+lib
 fixtures
 coverage
 __snapshots__
-src/generated
+generated
```
Comment (adidahiya) [inline .eslintignore:7]: why this change? are there other generated folders?

File: .npmrc (hunk 1 above)
Comment (justinbhopper) [reply .npmrc:1]: yarn respects this file too, but you are right it is not necessary assuming users have not overridden their default registry locally.  I had overridden mine, but have recently fixed that.  Will remove

File: .eslintignore (hunk 2 above)
Comment (justinbhopper) [reply .eslintignore:7]: For some reason, eslint was linting the generated folders.  I couldn't figure out why, because the path `src/generated` seems correct.  I had assumed it was due to some win-platform thing.  I will retest and see if I can avoid this change or improve the path.

Comment (justinbhopper) [reply .eslintignore]: I modified the path to be `**/src/generated` and it appears to work.  I can only surmise that `src/generated` was being interpretted as an absolute path from the root.  According to the docs and examples I found online, using ** for wildcard nested directories is necessary.

File: packages/eslint-config/index.js [hunk 3]
```diff
@@ -30,6 +30,7 @@ module.exports = {
     parser: "@typescript-eslint/parser",
     parserOptions: {
         sourceType: "module",
+        createDefaultProgram: true,
```
Comment (adidahiya) [inline packages/eslint-config/index.js:33]: this ended up causing a huge performance hit (see https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser#parseroptionscreatedefaultprogram), so I am reverting this change in https://github.com/palantir/blueprint/pull/4078
